### ISSUE TITLE
Link issues in namepaths with special characters

### ISSUE THREAD

**Issue by hegemonic** [jsdoc3/jsdoc#396]
The namepath docs include the following example of code that uses special characters in namepaths:

```js
/** @namespace */
var chat = {
    /**
     * Refer to this by {@link chat."#channel"}.
     * @namespace
     */
    "#channel": {
        /**
         * Refer to this by {@link chat."#channel".open}.
         * @type {boolean}
         * @defaultvalue
         */
        open: true,
        /**
         * Internal quotes have to be escaped by backslash. This is
         * {@link chat."#channel"."say-\"hello\""}.
         */
        'say-"hello"': function (msg) {}
    }
};

/**
 * Now we define an event in our {@link chat."#channel"} namespace.
 * @event chat."#channel"."op:announce-motd"
 */
```

Unfortunately, the methods in `templateHelper` aren't so kind to these namepaths:

+ The generated file names are just underscores. We should probably strip quotation marks--being careful about escaped quotation marks!--before we truncate the output file name.
+ Links to specific members are broken. I haven't investigated why.

This is a somewhat unusual case, but we should handle it better.

**Comment by hegemonic** [comment]
We now generate reasonable filenames for this scenario. However, there are still several problems:

+ We handle `chat."#channel"` incorrectly. We change its longname to `chat#channel` and say that `channel` is an instance member of `chat`. I'm guessing this is a regression in 3.3.0, although I could be wrong.
+ The event `chat."#channel".event:"op:announce-motd"` does not appear in the docs for `chat."#channel"` (probably because we mangle the longname for `chat."#channel"`).
+ In the navbar, the link to the event points to the file `chat.__channel_.html`, which doesn't exist (because we mangle the longname for `chat."#channel"`).

**Comment by hegemonic** [comment]
Looks like I broke the `chat."#channel"` handling in 44d9ec6831e498bd082de98a5bc60abfdba2bb7f, which is the "new parser infrastructure" mega-commit. That'll be fun to track down...

**Comment by hegemonic** [comment]
I filed #950 for the messed-up namepath.

**Comment by hegemonic** [comment]
The fix for #950 appears to have resolved all of the issues mentioned above. Closing.
